embassy-net/src/lib.rs

```diff
@@ -75,10 +68,34 @@ pub struct Config {
     pub dns_servers: Vec<Ipv4Address, 3>,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct DhcpConfig {
+    pub max_lease_duration: Option<Duration>,
+    pub retry_config: RetryConfig,
+    /// Ignore NAKs.
+    pub ignore_naks: bool,
+    /// Server port config
+    pub server_port: u16,
+    /// Client port config
+    pub client_port: u16,
+}
+
+impl Default for DhcpConfig {
+    fn default() -> Self {
+        Self {
+            max_lease_duration: Default::default(),
+            retry_config: Default::default(),
+            ignore_naks: Default::default(),
+            server_port: smoltcp::wire::DHCP_SERVER_PORT,
+            client_port: smoltcp::wire::DHCP_CLIENT_PORT,
+        }
+    }
+}
+
 pub enum ConfigStrategy {
```

Comment:
Now that there's config structs for DHCP and static it makes sense to rename like this so they're more consistent:

Config -> StaticConfig
ConfigStrategy -> Config